### PR TITLE
Update value of `rng_salt` to 0.42.0

### PR DESCRIPTION
**Pull request by JerryChen97**
------------------------------------------------------------------------------------------------------------

**Context:**
Set this ready for review after 0.42 released

**Description of the Change:**
See https://github.com/PennyLaneAI/pennylane/pull/7727#issuecomment-3084849955 for a detailed list of the 9 fails this time.
TL;DR
Some are shot vector tests which should remove accuracy validations as last time; one xfailed now passed, and one fluke; others are repeating local-salted ones, which requrie separate PR's to be revised and fixed.
For now, we will change their salts.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**[sc-93829]

### CODE REVIEW

**Comment by JerryChen97** [comment]
Close it for now since it's still quite far away from it's getting into effect (after next release fully concluded)

**Comment by JerryChen97** [comment]
Note: this PR will be WIP even being `ready-for-review`, only for triggering the CI. It won't be **truly** ready-for-review until it's fully discussed internally.

**Comment by JerryChen97** [comment]
# Tests that failed this update:

1st Round:
 - [x] `tests/devices/qubit/test_simulate.py::TestMidMeasurements::test_simulate_one_shot_native_mcm[pad-invalid-samples-torch]`
 - [x] **NEW!** `tests/measurements/test_probs.py::TestProbs::test_integration_jax[params1-obs1-500]`
 - [x] `tests/capture/test_capture_mid_measure.py::TestMidMeasureExecute::test_simple_circuit_execution[phi1-None-False-var-100]`
 - [x] `tests/devices/qubit/test_dq_interpreter.py::TestSampling::test_projector_samples_hw_like[1]`
 - [x] `tests/test_debugging.py::TestSnapshotUnsupportedQNode::test_lightning_qubit_finite_shots`
 - [x] Fluke. Pass now. ~**NEW!** `tests/transforms/test_qcut.py::TestCutCircuitMCTransform::test_cut_circuit_mc_expval[DefaultQubit]`~
 - [x] *XFAILD NOW PASS*`tests/gradients/finite_diff/test_spsa_gradient.py::TestRademacherSampler::test_mean_and_var[10000-ids0-5]`

2nd Round:
 - [x] *KNOWN ISSUE* `tests/workflow/interfaces/qnode/test_jax_jit_qnode.py::TestShotsIntegration::test_shot_vectors_single_measurements[shots0-jax]`
 - [x] *KNOWN ISSUE* `tests/workflow/interfaces/qnode/test_jax_jit_qnode.py::TestShotsIntegration::test_shot_vectors_multiple_measurements[shots1-auto]`

File: tests/devices/qubit/test_simulate.py [hunk 1]
```diff
@@ -1592,7 +1592,8 @@ def test_tree_traversal_combine_measurements(self, measurements, expected):
         else:
             assert qml.math.allclose(combined_measurement, expected)
 
-    @pytest.mark.local_salt(8)
+    # Near 10% failure rate; need revise and fix soon
```
Comment (andrijapau) [inline tests/devices/qubit/test_simulate.py:1595]: Can we change this to a `FIX` or `TODO` comment with a story?

File: tests/test_debugging.py [hunk 2]
```diff
@@ -632,7 +632,10 @@ def circuit():
         with pytest.warns(UserWarning, match="Snapshots are not supported"):
             _ = qml.snapshots(circuit)
 
-    @pytest.mark.local_salt(1)
+    # Improper ttest_ind usage, but leaving it here for now;
```
Comment (andrijapau) [inline tests/test_debugging.py:641]: Same here

File: tests/devices/qubit/test_simulate.py (hunk 1 above)
Comment (JerryChen97) [reply tests/devices/qubit/test_simulate.py:1595]: Good idea. Working on it now

File: tests/capture/test_capture_mid_measure.py [hunk 3]
```diff
@@ -322,7 +322,7 @@ class TestMidMeasureExecute:
 
     # NOTE: this test has an estimated fail rate of around 20%~30%
     # We have to fix the seed to ensure that the test is deterministic.
```
Comment (JerryChen97) [inline tests/capture/test_capture_mid_measure.py]:
```suggestion
    # FIXME: [sc-95723]
```

File: tests/devices/qubit/test_dq_interpreter.py [hunk 4]
```diff
@@ -368,7 +368,8 @@ def f():
         s2 = f()  # should be done with different key, leading to different results.
         assert not qml.math.allclose(s1, s2)
 
-    @pytest.mark.local_salt(6)
+    # 20 % failure rate; need to revise and fix soon
```
Comment (JerryChen97) [inline tests/devices/qubit/test_dq_interpreter.py:371]:
```suggestion
    # 20 % failure rate; need to revise and fix soon
    # FIXME: [sc-95722]
```

File: tests/devices/qubit/test_simulate.py (hunk 1 above)
Comment (JerryChen97) [inline tests/devices/qubit/test_simulate.py:1595]:
```suggestion
    # Near 10% failure rate; need revise and fix soon
    # FIXME: [sc-95724]
```

File: tests/test_debugging.py [hunk 5]
```diff
@@ -632,7 +632,10 @@ def circuit():
...
+    # should be revised and fixed soon
+    # current failure rate: ~7%
```
Comment (JerryChen97) [inline tests/test_debugging.py:643]:
```suggestion
    # current failure rate: ~7%
    # FIXME: [sc-92966]
```

File: tests/test_debugging.py (hunk 2 above)
Comment (JerryChen97) [reply tests/test_debugging.py:641]: done

File: tests/devices/qubit/test_simulate.py (hunk 1 above)
Comment (JerryChen97) [reply tests/devices/qubit/test_simulate.py:1595]: done

**Comment by JerryChen97** [comment]
> Looks pretty standard, for the ones that are salted are we sure those aren't magic seeds?

That's exactly the point of our seed-salt system: those salted twice now need to be watched, revised and better fixed in separate stories.

**Comment by JerryChen97** [comment]
@andrijapau @astralcai

File: tests/measurements/test_probs.py [hunk 6]
```diff
@@ -336,6 +336,7 @@ def circuit():
         expected = np.array([0.5, 0.5, 0, 0])
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    @pytest.mark.local_salt(2)
```
Comment (andrijapau) [inline tests/measurements/test_probs.py]: Does this need a story?

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py [hunk 7]
```diff
@@ -878,7 +878,7 @@ def cost_fn(a, b):
 
     @pytest.mark.parametrize("shots", [(10000, 10000), (10000, 10005)])
     def test_shot_vectors_single_measurements(self, interface, shots, seed):
```
Comment (andrijapau) [inline tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:922]: Does this have a corresponding accuracy test somewhere else?

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py [hunk 8]
```diff
@@ -889,18 +889,24 @@ def circuit(x):
...
-        assert qml.math.allclose(g[1], expected_g, atol=1 / qml.math.sqrt(shots[1]), rtol=0.03)
+
+        # Test gradient shapes for shot vectors
+        assert isinstance(g, tuple)
+        assert len(g) == 2  # Two different shot counts
+        assert all(isinstance(gr, jax.numpy.ndarray) for gr in g)
+        assert all(gr.shape == () for gr in g)  # Scalar gradients
 
     @pytest.mark.parametrize("shots", [(10000, 10000), (10000, 10005)])
     def test_shot_vectors_multiple_measurements(self, interface, shots, seed):
```
Comment (andrijapau) [inline tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:950]: and this?

**Review by andrijapau — COMMENTED** [review]
Just some final questions

File: tests/measurements/test_probs.py (hunk 6 above)
Comment (JerryChen97) [reply tests/measurements/test_probs.py]: I'll add one

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py (hunk 7 above)
Comment (JerryChen97) [reply tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:922]: I actually don't see. I'll add one for them

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py (hunk 8 above)
Comment (JerryChen97) [reply tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:950]: https://github.com/PennyLaneAI/pennylane/pull/7727/commits/705fef0f25da569d7f5bd51cde3c69e7d572ac34

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py (hunk 7 above)
Comment (JerryChen97) [reply tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:922]: https://github.com/PennyLaneAI/pennylane/pull/7727/commits/705fef0f25da569d7f5bd51cde3c69e7d572ac34

File: tests/measurements/test_probs.py (hunk 6 above)
Comment (JerryChen97) [reply tests/measurements/test_probs.py]: https://github.com/PennyLaneAI/pennylane/pull/7727/commits/9db1d69d16ecd67b4fb27bbe19618fb40a2b2913

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py [hunk 9]
```diff
@@ -912,20 +910,74 @@ def circuit(x):
...
-            res[1][1][0], expected_probs[0], atol=1 / qml.math.sqrt(shots[0]), rtol=0.03
+            res[1][0], expected_probs[0], atol=1 / qml.math.sqrt(shots), rtol=0.03
         )
         # Smaller atol since sin(0.25)**2 is close to zero
-        assert qml.math.allclose(
-            res[1][1][1], expected_probs[1], atol=0.5 * 1 / qml.math.sqrt(shots[1])
-        )
+        assert qml.math.allclose(res[1][1], expected_probs[1], atol=0.5 * 1 / qml.math.sqrt(shots))
+
+    @pytest.mark.parametrize("shots", [(10000, 10000), (10000, 10005)])
```
Comment (andrijapau) [inline tests/workflow/interfaces/qnode/test_jax_jit_qnode.py]: Do we need this many shots if we are doing a shape test?

Comment (JerryChen97) [reply tests/workflow/interfaces/qnode/test_jax_jit_qnode.py]: https://github.com/PennyLaneAI/pennylane/pull/7727/commits/2fd97ab1d2d1513071a93446baaa7660bf389e4d

**Review by astralcai — COMMENTED** [review]
Can you also check if any of the existing local salts added from previous releases can be removed with this new global salt?

**Comment by JerryChen97** [comment]
@andrijapau @astralcai I just realize it's another common issue when testing with gradients: some `expected` can have components among which the dimensions vary a lot e.g. `[np.float64(0.26433581791313), np.float64(22.456499531730763)]` in `tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py::TestParameterShiftRule::test_involutory_and_noninvolutory_variance_single_param`
Under such circumstance it's not reasonable to use `atol` only. One quick fix could be apply `rtol` for such cases, but I believe among our gigantic testing code base it will be a highly non-trivial task. So I backlog a story for it. [sc-96170]

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 10]
```diff
@@ -1187,7 +1187,7 @@ def test_prob_expectation_values(self, broadcast):
             assert isinstance(r_to_check, np.ndarray)
             assert r_to_check.shape == (4,)
 
-    @pytest.mark.local_salt(42)
+    # @pytest.mark.local_salt(42)
```
Comment (astralcai) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
```

File: tests/devices/qubit/test_sampling.py [hunk 11]
```diff
@@ -986,7 +986,7 @@ def test_sample_measure(self, mocker, seed):
     # but the definition of `atol` is too arbitrary. Further
     # investigation is needed to establish a more systematic
     # approach to test the final sampling distribution. [sc-91887]
-    @pytest.mark.local_salt(2)
+    # @pytest.mark.local_salt(2)
```
Comment (astralcai) [inline tests/devices/qubit/test_sampling.py]:
```suggestion
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 12]
```diff
@@ -1226,7 +1226,7 @@ def test_involutory_variance_single_param(self, broadcast):
         for _gA in gradA:
             assert _gA == pytest.approx(expected, abs=shot_vec_tol)
 
-    @pytest.mark.local_salt(42)
+    # @pytest.mark.local_salt(42)
```
Comment (astralcai) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 13]
```diff
@@ -1315,7 +1315,7 @@ def test_non_involutory_variance_single_param(self, broadcast, seed):
             assert gradF.shape == ()
             assert qml.math.allclose(gradF, expected, atol=2 * _herm_shot_vec_tol)
 
-    @pytest.mark.local_salt(42)
+    # @pytest.mark.local_salt(42)
```
Comment (astralcai) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 14]
```diff
@@ -1432,7 +1432,7 @@ def test_involutory_and_noninvolutory_variance_single_param(self, broadcast, see
             assert qml.math.allclose(shot_vec_result[0], expected[0], atol=0.1)
             assert qml.math.allclose(shot_vec_result[1], expected[1], atol=1.5)
 
-    @pytest.mark.local_salt(42)
+    # @pytest.mark.local_salt(42)
```
Comment (astralcai) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
```

File: tests/templates/test_subroutines/test_reflection.py [hunk 15]
```diff
@@ -164,7 +164,7 @@ def test_lightning_qubit(self):
 
     # NOTE: the finite shot test of the results has a 3% chance to fail
     # due to the random nature of the sampling. Hence we just pin the salt
-    @pytest.mark.local_salt(1)
+    # @pytest.mark.local_salt(1)
```
Comment (astralcai) [inline tests/templates/test_subroutines/test_reflection.py]:
```suggestion
```

File: tests/templates/test_subroutines/test_reflection.py [hunk 16]
```diff
@@ -185,7 +185,7 @@ def test_qnode_autograd(self, shots, seed):
 
         assert np.allclose(res, self.exp_jac, atol=0.005)
 
-    @pytest.mark.local_salt(1)
+    # @pytest.mark.local_salt(1)
```
Comment (astralcai) [inline tests/templates/test_subroutines/test_reflection.py]:
```suggestion
```

File: tests/templates/test_subroutines/test_reflection.py [hunk 17]
```diff
@@ -218,7 +218,7 @@ def test_qnode_jax(self, shots, use_jit, seed):
 
         assert np.allclose(jac, self.exp_jac, atol=0.005)
 
-    @pytest.mark.local_salt(1)
+    # @pytest.mark.local_salt(1)
```
Comment (astralcai) [inline tests/templates/test_subroutines/test_reflection.py]:
```suggestion
```

**Review by astralcai — APPROVED** [review]
Thank you for staying on top of this and making sure that every issue is tracked! 🚀

**Review by andrijapau — APPROVED** [review]
Nice! Another one down 😎
